dbms/src/Databases/DatabaseFactory.h

```diff
@@ -2,6 +2,7 @@
 
 #include <Common/ThreadPool.h>
 #include <Databases/IDatabase.h>
+#include <Parsers/ASTCreateQuery.h>
```

Comment:
It's not necessarily to include header, because you can use forward declaration.